test/test_framework_test.go

```diff
@@ -297,16 +297,30 @@ func TestExecuteScript(t *testing.T) {
 func TestImportContract(t *testing.T) {
 	t.Parallel()
 
-	t.Run("init no params", func(t *testing.T) {
+	t.Run("contract with no init params", func(t *testing.T) {
 		t.Parallel()
 
 		const code = `
             import Test
             import FooContract from "./FooContract"
 
+            pub let blockchain = Test.newEmulatorBlockchain()
+
+            pub fun init() {
```

Comment:
We already have the `setup` special function, which is run once before all test functions. But I chose to introduce a new one, `init`, in order not to cause any undesired behavior for current test suites. Not a very strong opinion on this though.

Reply:
Makes sense, I have seen the names `beforeAll` / `afterAll` and `beforeEach` / `afterEach`. We already make use of the last pair, so I guess we can also replace `setup` / `tearDown` with `beforeAll` / `afterAll`

Reply:
I got rid entirely of the `init` special function, since we are going to use the mapping from `flow.json`. This means that deployments can take place in `setup`. I wanted to keep the breaking changes to a minimum :sweat_smile:

Reply:
nice!